BE/app/Traits/CacheExternalDataTrait.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Traits;
+
+use Illuminate\Support\Facades\Cache;
+
+/**
+ * Trait CacheExternalDataTrait
+ * @package App\Traits
+ */
+trait CacheExternalDataTrait
+{
+    protected static array $countries = [
+        'United States',
+        'United Kingdom',
+        'Australia',
+        'New Zealand'
+    ];
+
+    /**
+     * @param array $data
+     * @return bool
+     */
+    protected function cacheExistsData(array $data): bool
+    {
+        if (!empty($data)) {
+            $dataByCountries = array_filter($data, function ($data) {
+                return !empty($data['country']) && in_array($data['country'], self::$countries);
+            });
+
+            Cache::forever('universities', $dataByCountries);
```

Comment:
Cache TTL should be configurable

Reply:
I missed point in the task where it was written about cache ttl, and I used forever instead Cache::put(‘cache_key’, $data, now()->addDays(2))

Reply:
That would be better but it also needs to be configurable